packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -416,3 +433,72 @@ const toChatError = (target: ChatErrorTarget, error: unknown): ChatError => {
   }
   return new ChatError(target, new Error(`${error}`));
 };
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+class MessageQueue {
+  private _messageQueue: ChatMessageWithStatus[] = [];
+  private _context: ChatContext;
+  private _credential: CommunicationTokenCredential;
+
+  constructor(context: ChatContext, credential: CommunicationTokenCredential) {
+    this._context = context;
+    this._credential = credential;
+  }
+
+  public containsMessage(message: ChatMessageWithStatus): boolean {
+    let contains = false;
+    if (this._messageQueue.includes(message)) {
+      contains = true;
+    }
+    return contains;
+  }
+
+  public async addMessage(threadId: string, message: ChatMessageWithStatus): Promise<void> {
+    // make a copy of message and add to queue
+    const copy = { ...message };
+    this._messageQueue.push(copy);
+    const newMessage = await this.requestAttachments();
+    if (newMessage) {
```

Comment:
Does the message queue logic need to be implemented here?
AddMessage = add to the queue, start queue if not running.

Queue should then download attachments serially in order of the queue (FIFO).

Queue updates messages as they have local cached resources.

Reply:
Introduce a new method to start queue